universe/ValueRefs.h

```diff
@@ -196,6 +211,14 @@ struct FO_COMMON_API ComplexVariable final : public Variable<T>
     const ValueRef<std::string>* StringRef2() const;
     unsigned int GetCheckSum() const override;
 
+    std::unique_ptr<ValueRef<T>> Clone() const override {
+        return std::make_unique<ComplexVariable<T>>(Variable<T>::PropertyName()[0],
```

Comment:
Can't use `operator[]` on the return result of `PropertyName()` because `operator[]` can mutate what it operates on and `PropertyName()` returns a const reference. Also not sure if that will really fully clone `this` in all cases...

Reply:
These non-trivial `Clone()` implementations should probably be in the .cpp not the .h

Reply:
It's a template so it cannot be in *.cpp. 

`ComplexVariable` constructor fills single variable name as I understand so PropertyName here should have only one value.

Reply:
> It's a template so it cannot be in *.cpp.

good point

> `ComplexVariable` constructor fills single variable name as I understand so PropertyName here should have only one value.

perhaps, but no reason to not just pass the whole `vector`, and a couple other reasons to do so.